src/kiwixapp.cpp

```diff
@@ -93,11 +93,20 @@ void KiwixApp::init()
         }
     });
     connect(&m_watcher, &QFileSystemWatcher::directoryChanged, this, [=](QString monitorDir) {
-        m_library.loadMonitorDirInThread(monitorDir);
+        m_library.asyncLoadMonitorDir(monitorDir);
     });
     QString monitorDir = KiwixApp::instance()->getSettingsManager()->getMonitorDir();
+    QStringList zimsInMonitorDir;
+    for (auto str : m_library.getBookIds()) {
+        auto filePath = QString::fromStdString(m_library.getBookById(str).getPath());
+        QDir absoluteDir = QFileInfo(filePath).absoluteDir();
+        if (absoluteDir == monitorDir) {
+            zimsInMonitorDir.push_back(filePath);
+        }
+    }
```

Comment:
Please make this a member function of `Library` (and the list of zim files from the monitored directory should be a data-member of `Library` too).